### PR TITLE
[FLOC-3999] Documentation redirects.

### PR DESCRIPTION
**Pull request by tomprince**
https://clusterhq.atlassian.net/browse/FLOC-3999

### CODE REVIEW

File: admin/test/test_release.py [hunk 1]
```diff
@@ -63,13 +70,123 @@ def hard_linking_possible():
...
 
 
+def MatchesRoutingRules(rules):
+    """
+    Matches against routing rules.
+
+    :param rules: The routing rules to match against.
+    :type rules: ``list`` of ``RoutingRule``
+    """
+    return AfterPreprocessing(RoutingRules.to_xml,
```
Comment (itamarst) [inline admin/test/test_release.py]: This looks like it'll always be true?

Comment (itamarst) [inline admin/test/test_release.py]: Ah, no, I see.

File: docs/redirects.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"en/latest/":
+    # Redirects in /latest/ introduced for 1.10.0 doc reorganization
+    "introduction/": {replace_key: ""}
+    "introduction/flocker-supports.html": {replace_key: "supported/"}
```
Comment (itamarst) [inline docs/redirects.yaml]: So this means `docs.clusterhq.com/en/latest/introduction/flocker-supports.html` gets redirected to `docs.clusterhq.com/en/latest/supported/`? And I see `http://doc-dev.clusterhq.com/supported/` exists and that makes sense. OK.

File: docs/redirects.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "concepts/": {replace_key: "flocker-features/"}
+    "labs/installer.html": {replace_key: "flocker-standalone/installer.html"}
+    "labs/volumes-cli.html": {replace_key: "flocker-features/flockerctl.html"}
+    "labs/volumes-gui.html": {hostname: "clusterhq.com", replace_key: "volumehub/"}
+    "labs/swarm.html": {replace_key: "docker-integration/"}
+    "labs/compose.html": {replace_key: "docker-integration/"}
+    "labs/mesosphere.html": {replace_key: "mesos-integration/"}
+    "labs/kubernetes.html": {replace_key: "kubernetes-integration/"}
+    "labs/flockercontainers.html": {replace_key: "supported/flockercontainers.html"}
+    "labs/docker-plugin.html": {replace_key: "docker-integration/"}
```
Comment (itamarst) [inline docs/redirects.yaml]: docs.clusterhq.com/en/latest/labs/docker-plugin.html is a 404.  Maybe "introduction/flocker-plugin.html" though?

File: docs/redirects.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+"en/latest/":
```
Comment (itamarst) [inline docs/redirects.yaml]: So I wonder about this... Currently `docs.clusterhq.com/en/latest/` immediately redirects to `docs.clusterhq.com/en/1.9.0`. That means external people are not going to be linking to it, mostly, but rather to whatever version of docs they randomly found. And new visitors will get warning and that's OK I guess. So is this mostly for internal links from our marketing material/website?

**Comment by itamarst** [comment]
So skimming this it looks vaguely correct, but the proof of the pudding is in the eating. I guess live test as part of release will verify this? So address the one minor actionable comment and merge when you're happy.

File: docs/redirects.yaml (hunk 4 above)
Comment (tomprince) [inline docs/redirects.yaml]: Once 1.10.0 is released, the 1.9.0 will have a link to the corresponding page in latest, which doesn't currently exist, but this introduces a redirect for it.

File: docs/redirects.yaml (hunk 3 above)
Comment (lukemarsden) [inline docs/redirects.yaml]: Yes, but it wasn't previously. We have had error reports from users landing on this page -- the link must be out there somewhere. We should add `introduction/flocker-plugin.html` as a link to there as well though.

File: docs/redirects.yaml (hunk 4 above)
Comment (lukemarsden) [inline docs/redirects.yaml]: @dwgebler's recent change in https://github.com/ClusterHQ/flocker/pull/2503 makes `/en/latest` stop being a redirect, and instead be the canonical location for the docs, fwiw.

File: docs/redirects.yaml (hunk 3 above)
Comment (tomprince) [inline docs/redirects.yaml]: For reference, this existed in 1.4.0.

**Pull request merged**